### PR TITLE
Add new runner variable `runner.arch`

### PR DESCRIPTION
**Pull request by fhammerl**
### Why:

Adding `runner.arch` variable after https://github.com/actions/runner/issues/1185 is merged.

### What's being changed:

Context:

https://gha-docs-10389--add-runner-arc.herokuapp.com/en/actions/learn-github-actions/contexts#runner-context

Environment variable:

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Duplicate of #

**Comment by Vendetta26** [comment]
> Thanks for your submission.
> 
> It appears that you've created a pull request using one of our repository's branches. Since this is almost always a mistake, we're going to go ahead and close this. If it was intentional, please let us know what you were intending and we can see about reopening it.
> 
> Thanks again!

**Review by ghost — COMMENTED** [review]
Duplicate of #

**Comment by Wiatt99923** [comment]
Need my commission account to start paying me

**Comment by ramyaparimi** [comment]
@fhammerl 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by lucascosti — DISMISSED** [review]
Thanks a lot for this PR, @fhammerl!

Before we merge it, I just wanted to double check: because these new variables are set by the `runner` app, I assume that this change isn't tied to specific GHES versions? i.e. a self-hosted runner attached to a GHES server will be able to use this environment variable after the `runner` app is next automatically updated on the runner machine, regardless of the GHES instance version?

**Comment by fhammerl** [comment]
> Thanks a lot for this PR, @fhammerl!
> 
> Before we merge it, I just wanted to double check: because these new variables are set by the `runner` app, I assume that this change isn't tied to specific GHES versions? i.e. a self-hosted runner attached to a GHES server will be able to use this environment variable after the `runner` app is next automatically updated on the runner machine, regardless of the GHES instance version?

@lucascosti ☝️ that's correct, this is set in the runner. The change was shipped in runner v2.284.0. (afaik GHES 3.3 currently uses v2.283.3)
If a self-hosted runner (regardless of GHES or dotcom version) is that version or higher, `runner.arch` is available.

**Comment by lucascosti** [comment]
> afaik GHES 3.3 currently uses v2.283.3

ah, ok. In that case, I'll add some versioning to make this available to GHES versions 3.4 and above 👍

File: data/features/actions-runner-arch-envvars.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Reference: https://github.com/github/docs/pull/12374
```
Comment (lucascosti) [inline data/features/actions-runner-arch-envvars.yml:1]:
```suggestion
# Reference: https://github.com/github/docs-content/issues/5727
```

File: data/features/actions-runner-arch-envvars.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Reference: https://github.com/github/docs-content/issues/5727
```
Comment (lucascosti) [inline data/features/actions-runner-arch-envvars.yml:1]:
```suggestion
# Reference: #5727
```

**Review by martin389 — APPROVED** [review]
Looks good in staging: 👍 
- [enterprise-cloud](https://docs-12374--add-runner-arch-va.herokuapp.com/en/enterprise-cloud@latest/actions/learn-github-actions/contexts)
- [FPT](https://docs-12374--add-runner-arch-va.herokuapp.com/en/actions/learn-github-actions/contexts#runner-context)

**Pull request merged**